### ISSUE TITLE
Show dev channel deprecation infobar

### ISSUE THREAD

**Issue by simonhong** [brave/brave-browser#33822]
As we'll deprecate dev channel soon, launch infobar about this.

Related issue: https://github.com/brave/brave-browser/issues/25248

**Comment by stephendonner** [comment]
@simonhong can we get an uplift going for this to `beta`, so it can be verified using `dev`?  Thanks!

**Comment by kjozwiak** [comment]
The above requires `1.61.75` or higher for `1.61.x` verification. We'll need to check this on all platforms and ensure that the message is appearing on `Dev` for `Win`, `macOS` & `Linux`. We'll also need to make sure that the above isn't appearing on the `BETA` version of Brave once we have builds with the above fix.

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave	1.61.79 
```

and

```
Brave	1.61.83 
```

<details>
<summary>Dev Channel - InProgress</summary>

<details>
<summary>Clean profile - InProgress</summary>

1. Installed `1.61.79` version (dev channel)
2. Confirmed on first launch the deprecation infobar was shown
3. Confirmed the "Learn more" link directed to a support page (https://support.brave.com/hc/en-us/articles/17924707453581-How-do-I-migrate-my-Brave-Dev-data-to-another-channel-Nightly-Beta-Release-)
4. Confirmed clicking on the "x" closes the infobar
5. Confirmed able to navigate normally in the browser (opened a few sites)
6. Confirmed after closing and relaunching the browser, the infobar displayed again on relaunch.

Step 2 | Step 3 | Step 4 | Step 6
--- | --- | --- | ---

</details>

<details>
<summary>Upgrade profile - TBD</summary>

1. Installed `1.61.71` version (dev channel before the fix)
2. Launched and opened a few tabs
3. Closed and upgraded to `1.61.79` (version with the fix)
4. Confirmed on launch the deprecation infobar was shown
5. Confirmed the "Learn more" link directed to a support page (https://support.brave.com/hc/en-us/articles/17924707453581-How-do-I-migrate-my-Brave-Dev-data-to-another-channel-Nightly-Beta-Release-)
6. Confirmed clicking on the "x" closes the infobar
7. Confirmed able to navigate normally in the browser (opened a few sites)
8. Confirmed after closing and relaunching the browser, the infobar displayed again on relaunch.

Step 2 | Step 4 | Step 5 | Step 6 | Step 8 
--- | --- | --- | --- | ---

</details>

</details>

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on

```
Brave | 1.61.83 Chromium: 119.0.6045.163 (Official Build) beta (64-bit)
-- | --
Revision | f5a497784a959e96317dc946d9086098d8fa6c50
OS | Windows 10 Version 22H2 (Build 19045.3693)
```
<details>
<summary>Dev Channel - </summary>

<details>
<summary>Clean profile - </summary>

1. Installed `1.61.79` version (dev channel)
2. Confirmed on first launch the deprecation infobar was shown
3. Confirmed the "Learn more" link directed to a support page (https://support.brave.com/hc/en-us/articles/17924707453581-How-do-I-migrate-my-Brave-Dev-data-to-another-channel-Nightly-Beta-Release-)
4. Confirmed clicking on the "x" closes the infobar
5. Confirmed able to navigate normally in the browser (opened a few sites)
6. Confirmed after closing and relaunching the browser, the infobar displayed again on relaunch.

Step 2 | Step 3 | Step 4 | Step 6
--- | --- | --- | ---
<img width="1202" alt="2" src="https://github.com/brave/brave-browser/assets/28145373/3cc65efd-94f6-463f-a268-7e0a1e17f557"> | <img width="1214" alt="3" src="https://github.com/brave/brave-browser/assets/28145373/b74ce879-8b4a-4387-93f8-a38b5452f0fd"> | <img width="1207" alt="4" src="https://github.com/brave/brave-browser/assets/28145373/be555b10-cddb-4fd6-b8e6-86e5a63dadc3"> | <img width="1207" alt="6" src="https://github.com/brave/brave-browser/assets/28145373/6aa15a0d-8ac7-48c1-93fb-73e093b6501a">

</details>

<details>
<summary>Upgrade profile - </summary>

1. Installed `1.61.71` version (dev channel before the fix)
2. Launched and opened a few tabs
3. Upgraded to `1.61.79` (version with the fix)
4. Relaunched and confirmed on launch the deprecation infobar was shown
5. Confirmed the "Learn more" link directed to a support page (https://support.brave.com/hc/en-us/articles/17924707453581-How-do-I-migrate-my-Brave-Dev-data-to-another-channel-Nightly-Beta-Release-)
6. Confirmed clicking on the "x" closes the infobar
7. Confirmed able to navigate normally in the browser (opened a few sites)
8. Confirmed after closing and relaunching the browser, the infobar displayed again on relaunch.

Step 2 | Step 3 | Step 4 | Step 5 | Step 6 | Step 8 
--- | --- | --- | --- | --- | ---
<img width="1204" alt="2" src="https://github.com/brave/brave-browser/assets/28145373/3904e552-06fe-4355-967f-81cedf0e876b"> | <img width="1205" alt="3" src="https://github.com/brave/brave-browser/assets/28145373/9f3e5a74-6e80-4d22-af0c-437d2507796c"> | <img width="1206" alt="4" src="https://github.com/brave/brave-browser/assets/28145373/e7cce4b6-c67a-420b-96d0-c95ee8ee3521"> | <img width="1204" alt="5" src="https://github.com/brave/brave-browser/assets/28145373/19c945ea-46bc-418e-b0ad-e8c8e785c768"> | <img width="1206" alt="6" src="https://github.com/brave/brave-browser/assets/28145373/e12a1ddb-fd2d-4bd4-8e90-6305bed41c67"> | <img width="1203" alt="8" src="https://github.com/brave/brave-browser/assets/28145373/2eeeb8da-7ad8-40c4-86ec-43a2c7a3798f">

</details>

</details>



<details>
<summary>Beta Channel - </summary>

<details>
<summary>Clean profile - </summary>

1. Installed `1.61.83` version (beta channel)
2. Confirmed on first launch the deprecation infobar was NOT shown
3. Confirmed able to navigate normally in the browser (opened a few sites)
4. Confirmed after closing and relaunching the browser, the infobar is NOT displayed on relaunch

Step 2 | Step 4
--- | ---
<img width="1204" alt="2" src="https://github.com/brave/brave-browser/assets/28145373/ff112efb-0a7e-4cd7-ba08-943bac00b2c1"> | <img width="1202" alt="4" src="https://github.com/brave/brave-browser/assets/28145373/a3b483e6-7782-4164-b3a1-00e5bb210b10">

</details>

<details>
<summary>Upgrade profile - </summary>

1. Installed `1.61.70` version (beta channel before this PR)
2. Launched and opened a few tabs
3. Upgraded to `1.61.83` (version with this PR)
4. Confirmed on relaunch the deprecation infobar was NOT shown
5. Confirmed able to navigate normally in the browser (opened a few sites)
6. Confirmed after closing and relaunching the browser, the infobar is NOT displayed on relaunch.

Step 3 | Step 4 | Step 6
--- | --- | ---
<img width="1212" alt="3" src="https://github.com/brave/brave-browser/assets/28145373/4948e572-5372-4a4b-a79b-2e019ef87134"> | <img width="1211" alt="4" src="https://github.com/brave/brave-browser/assets/28145373/5a852fbd-ebd4-4f70-8c0f-17c7219c3e1b"> | <img width="1208" alt="6" src="https://github.com/brave/brave-browser/assets/28145373/e0e2a209-cd02-47db-b732-1297408eeaf1">

</details>

</details>
